x/crosschain/types/legacy_msgs.go

```diff
@@ -0,0 +1,152 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/node/pkg/authz"
+)
+
+// MsgVoteOnObservedInboundTx
+
+var _ sdk.Msg = &MsgVoteOnObservedInboundTx{}
+
+func (msg *MsgVoteOnObservedInboundTx) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgVoteOnObservedInboundTx) Type() string {
+	return authz.InboundVoter.String()
+}
+
+func (msg *MsgVoteOnObservedInboundTx) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgVoteOnObservedInboundTx) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgVoteOnObservedInboundTx) ValidateBasic() error {
+	return nil
```

Comment:
Should we return `err not supported` here ?
I am guess we don't need to execute the messages to be able to parse them on the indexer (might be wrong tough)

Reply:
It's just implemented to implement the Msg interface. The message can't be processed if not added explicitly in the MsgService so I think it's fine